domains/client/domain-operator/src/tests.rs

```diff
@@ -8171,6 +8183,91 @@ async fn test_current_block_number_used_as_new_account_nonce() {
     );
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_domain_node_starting_check() {
+    let directory = TempDir::new().expect("Must be able to create temporary directory");
+
+    let mut builder = sc_cli::LoggerBuilder::new("");
+    builder.with_colors(false);
+    let _ = builder.init();
+
+    let tokio_handle = tokio::runtime::Handle::current();
+
+    // Start Ferdie
+    let mut ferdie = MockConsensusNode::run(
+        tokio_handle.clone(),
+        Ferdie,
+        BasePath::new(directory.path().join("ferdie")),
+    );
+
+    // Run Alice (a evm domain authority node)
+    let alice = domain_test_service::DomainNodeBuilder::new(
+        tokio_handle.clone(),
+        BasePath::new(directory.path().join("alice")),
+    )
+    .build_evm_node(Role::Authority, Alice, &mut ferdie)
+    .await;
+
+    // Produce 5 consensus block with bundle
+    for _ in 0..5 {
+        ferdie.produce_block_with_extrinsics(vec![]).await.unwrap();
+    }
+    assert_eq!(ferdie.client.info().best_number, 5);
+    assert_eq!(alice.client.info().best_number, 0);
+
+    // Start one more domain node `Bob` on the same consensus node shoule be fine
+    let bob = domain_test_service::DomainNodeBuilder::new(
+        tokio_handle.clone(),
+        BasePath::new(directory.path().join("bob")),
+    )
+    .operator_id(1)
+    .build_evm_node(Role::Authority, Bob, &mut ferdie)
+    .await;
+
+    // Produce 3 domain blocks
+    produce_blocks!(ferdie, alice, 3, bob).await.unwrap();
+    assert_eq!(alice.client.info().best_number, 3);
+    assert_eq!(bob.client.info().best_number, 3);
+
+    // Stop `Bob`, produce more domain blocks, then restart `Bob` with the same
+    // consensus node should be fine
+    bob.stop().unwrap();
+    produce_blocks!(ferdie, alice, 3).await.unwrap();
+    assert_eq!(alice.client.info().best_number, 6);
+
+    let bob = domain_test_service::DomainNodeBuilder::new(
+        tokio_handle.clone(),
+        BasePath::new(directory.path().join("bob")),
+    )
+    .operator_id(1)
+    .build_evm_node(Role::Authority, Bob, &mut ferdie)
+    .await;
+    assert_eq!(bob.client.info().best_number, 3);
+
+    // It should be able catch up the missed domain block as more consensus block
+    // produced
+    ferdie.produce_block_with_extrinsics(vec![]).await.unwrap();
+    assert_eq!(bob.client.info().best_number, 6);
+
+    // Start another domain node with the same consensus node shoule be failed now
+    // because there are already domain block produced, the domain node has to start
+    // with a fresh consensus node
```

Comment:
There's actually a few typos here

```suggestion
    // Starting another domain node with the same consensus node should fail now.
    // Because there are already domain blocks produced, the domain node has to start
    // with a fresh consensus node
```